wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h

```diff
@@ -202,15 +108,305 @@ typedef wolfssl_TSIP_Hash wc_Sha;
     typedef struct {
         tsip_aes_key_index_t tsip_keyIdx;
         word32               keySize;
+        byte                 setup;
     } TSIP_AES_CTX;
+#endif 
+
+
+
+
+/****
+typedef struct user_PKCbInfo {
+    uint32_t user_key_id;
+} user_PKCbInfo;
+***/
+/*
+ * structure for ECDH key exchange
+ */
+typedef struct TsipUserCtx {
+    
+    uint32_t user_key_id;
+//    user_PKCbInfo *user_Cbinfo;
+#if (WOLFSSL_RENESAS_TSIP_VER >=109)     
+    /* out from R_SCE_TLS_ServerKeyExchangeVerify */
+    uint32_t encrypted_ephemeral_ecdh_public_key[ENCRYPTED_ECDHE_PUBKEY_SZ];
     
-    struct Aes;
-    int wc_tsip_AesCbcEncrypt(struct Aes* aes, byte* out, const byte* in,
-                              word32 sz);
-    int wc_tsip_AesCbcDecrypt(struct Aes* aes, byte* out, const byte* in,
-                              word32 sz);
+    /* ephemeral ECDH pubkey index 
+     * got from R_TSIP_GenerateTlsP256EccKeyIndex.
+     * Input to R_TSIP_TlsGeneratePreMasterSecretWithEccP256Key. 
+     */
+    tsip_tls_p256_ecc_key_index_t ecc_p256_wrapped_key;
+
+    /* ephemeral ECDH pub-key Qx(256bit)||Qy(256bit) 
+     * got from  R_TSIP_GenerateTlsP256EccKeyIndex.
+     * Should be sent to peer(server) in Client Key Exchange msg. 
+     */
+    uint8_t ecc_ecdh_public_key[ECCP256_PUBKEY_SZ];
+#endif /* WOLFSSL_RENESAS_TSIP_VER >=109 */
+
+    /* info to generate session key */
+    uint32_t    tsip_masterSecret[TSIP_TLS_MASTERSECRET_SIZE/4];
+    uint8_t     tsip_clientRandom[TSIP_TLS_CLIENTRANDOM_SZ];
+    uint8_t     tsip_serverRandom[TSIP_TLS_SERVERRANDOM_SZ];
+
+    /* TSIP defined cipher suite number */
+    uint32_t    tsip_cipher;    
     
-#endif /* WOLFSSL_RENESAS_TSIP_TLS_AES */
+    /* flags */
+    uint8_t pk_key_set:1;
+    uint8_t session_key_set:1;
+
+
+} TsipUserCtx;
+
+typedef TsipUserCtx user_PKCbInfo;
+
+typedef struct
+{
+    TsipUserCtx* userCtx; 
+} TsipPKCbInfo;
+
+
+#if (WOLFSSL_RENESAS_TSIP_VER >=109)
+
+typedef struct
+{
+    uint8_t *  encrypted_provisioning_key;
+    uint8_t *  iv;
+    uint8_t *  encrypted_user_tls_key;
+    uint32_t   encrypted_user_tls_key_type;
+    tsip_tls_ca_certification_public_key_index_t  user_rsa2048_tls_pubindex;
+} tsip_key_data;
+
+#else
+typedef struct
+{
+    uint8_t*  encrypted_session_key;
+    uint8_t*  iv;
+    uint8_t*  encrypted_user_tls_key;
+    tsip_tls_ca_certification_public_key_index_t  user_rsa2048_tls_pubindex;
+} tsip_key_data;
+
+#endif
+
+struct Aes;
+struct WOLFSSL;
+/*----------------------------------------------------*/
+/*   APIs                                             */
+/*----------------------------------------------------*/
+
+WOLFSSL_API void tsip_inform_cert_sign(const byte *sign);
+
+WOLFSSL_API void tsip_set_callbacks(struct WOLFSSL_CTX* ctx);
+
+WOLFSSL_API int  tsip_set_callback_ctx(struct WOLFSSL* ssl, void* user_ctx);
```

Comment:
Because part of this uses the `WOLFSSL` struct, it seems like we should add some `#ifndef` guards for `WOLFCRYPT_ONLY`.  Not added in this specific location, but in our TSIP port in general.  If a user builds the wolfSSL library with `WOLFCRYPT_ONLY` defined, there will not be a `WOLFSSL` struct available.  As such, TSIP TLS APIs will also not be available.

Reply:
Add macro statements in user_settings.h and settings.h so that  WOLFCRYPT_ONLY makes WOLFSSL_RENESAS_TSIP undefined.